src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,140 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private static final int INITIAL_SRC_POS = 0;
+    private static final double GROW_SIZE = 1.5;
+    private boolean arrayIsEmpty = true;
+    private T[] arrayData;
+    private int size;
+    private int currentSizeOfArray = INITIAL_SIZE;
+
+    @SuppressWarnings({"unchecked"})
+    public ArrayList() {
+        arrayData = (T[]) new Object[INITIAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (!arrayIsEmpty) {
+            if (ensureCapacity()) {
+                grow();
+            }
+            arrayData[size] = value;
+            size++;
+        } else {
+            arrayData[size] = value;
+            size++;
+            arrayIsEmpty = false;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (indexIsNegative(index) || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index are not exist, your index is "
+                    + index + " last index is " + size);
+        } else if (indexIsLegal(index) && ensureCapacity()) {
```

Comment:
you check index two times here. Lets make index check in indexIsLegal method. And we can throw ArrayListIndexOutOfBoundsException in indexIsLegal method, this will reduce the amount of duplicate code